drivers/audio/dmic_nrfx_pdm.c

```diff
@@ -540,11 +591,28 @@ static const struct _dmic_ops dmic_ops = {
 
 #define PDM_NRFX_DEVICE(idx)						     \
 	static void *rx_msgs##idx[DT_PROP(PDM(idx), queue_size)];	     \
-	static struct dmic_nrfx_pdm_drv_data dmic_nrfx_pdm_data##idx;	     \
+	static void event_handler##idx(const nrfx_pdm_evt_t *evt)	     \
+	{								     \
+		event_handler(DEVICE_DT_GET(PDM(idx)), evt);		     \
+	}								     \
+	PINCTRL_DT_DEFINE(PDM(idx));					     \
+	static const struct dmic_nrfx_pdm_drv_cfg dmic_nrfx_pdm_cfg##idx = { \
+		.event_handler = event_handler##idx,			     \
+		.pdm = NRFX_PDM_INSTANCE(idx),                               \
+		.nrfx_def_cfg =	NRFX_PDM_DEFAULT_CONFIG(0, 0),		     \
+		.nrfx_def_cfg.skip_gpio_cfg = true,			     \
+		.nrfx_def_cfg.skip_psel_cfg = true,			     \
+		.pcfg = PINCTRL_DT_DEV_CONFIG_GET(PDM(idx)),		     \
+		.clk_src = PDM_CLK_SRC(idx),				     \
+	};								     \
+	static struct dmic_nrfx_pdm_drv_data dmic_nrfx_pdm_data##idx =       \
+	{                                                                    \
+		.pdm = &dmic_nrfx_pdm_cfg##idx.pdm                           \
```

Comment:
There's no need to put the `nrfx_pdm_t` structure inside `struct dmic_nrfx_pdm_drv_cfg`, as it is anyway only accessed through the `dev->data` pointer. It can be a stand-alone `static const` variable instead, what will be less confusing IMO.
Also the static initialization is suboptimal in this case, because it requires that the whole `struct dmic_nrfx_pdm_drv_data` image is placed in flash, so that it can be copied to RAM at initialization. But you in fact need to initialize only one member, so it will be better to do it at runtime, in `pdm_nrfx_init##idx()`.
With these two corrections, also the diff in `PDM_NRFX_DEVICE(idx)` will be significantly smaller.